internal/processor/parser/app.go

```diff
@@ -170,11 +176,102 @@ func (am *AppParser) Parse(envelope *events.Envelope) ([]metric.MetricPackage, e
 	return metricsPackages, nil
 }
 
+// listApps is meant to replace the function from the go-cfclient and allow fetching apps in parallel tasks
+// The apps returned by this are just missing a reference to the client (unexported property) so don't use the Space() and Summary() methods
+func listApps(c *cfclient.Client, numWorkers int, log *gosteno.Logger) ([]cfclient.App, error) {
+
+	// Query the first page to get the total number of pages.
+	resp, err := queryAppsPage(c, 1)
+	if err != nil {
+		return nil, errors.Wrap(err, "Error requesting apps page 1, skipping cache warmup")
+	}
+	appResources := resp.Resources
+
+	var mutex sync.Mutex
+	var wg sync.WaitGroup
+
+	// Page 1 already fetched
+	pages := resp.Pages - 1
+	// No need for more workers than pages left
+	numWorkers = int(math.Min(float64(numWorkers), float64(pages)))
```

Comment:
if pages = 4 and numWorkers = 1 then numWorkers will be 0. Is that expected?
Same comment regarding pagesPerWorker.

Reply:
Hmm no we take the min between 4 and 1, so 1, and `numWorkers` will be 1. right ?

Reply:
I there is only 1 page though we will divide by 0. Fixing.

Reply:
Ah! i missed the min and the division by 0. Still something was off to me. I guess it's called instinct haha